agent/handlers/v2/task_container_stats_handler.go

```diff
@@ -37,16 +50,25 @@ func StatsHandler(state dockerstate.TaskEngineState, statsEngine stats.Engine) f
 			utils.WriteJSONToResponse(w, http.StatusBadRequest, errResponseJSON, utils.RequestTypeTaskStats)
 			return
 		}
-		if containerID := getContainerID(r.URL, StatsPath); containerID != "" {
-			writeContainerStatsResponse(w, taskARN, containerID, statsEngine)
+		if containerID := getStatsContainerID(r); containerID != "" {
+			seelog.Infof("V2 task/container stats handler: handling request for container '%s'", containerID)
+			WriteContainerStatsResponse(w, taskARN, containerID, statsEngine)
 			return
 		}
 
-		writeTaskStatsResponse(w, taskARN, state, statsEngine)
+		seelog.Infof("V2 task/container stats handler: handling request for task '%s'", taskARN)
+		WriteTaskStatsResponse(w, taskARN, state, statsEngine)
 	}
 }
 
-func writeTaskStatsResponse(w http.ResponseWriter,
+// getStatsContainerID gets the container ID for container stats
+func getStatsContainerID(r *http.Request) string {
```

Comment:
You can get rid of this function if you have a general method to retrieve from a request. See comments below.